source/adapters/level_zero/v2/queue_immediate_in_order.cpp

```diff
@@ -11,8 +11,51 @@
 #include "queue_immediate_in_order.hpp"
 
 namespace v2 {
+
+static int32_t getZeOrdinal(ur_device_handle_t Device, queue_group_type type) {
+  if (type == queue_group_type::MainCopy && Device->hasMainCopyEngine()) {
+    return Device->QueueGroup[queue_group_type::MainCopy].ZeOrdinal;
+  }
+  return Device->QueueGroup[queue_group_type::Compute].ZeOrdinal;
```

Comment:
If there's no copy engine available, we could have a special path (just a single CommandList and a single LastEvent in the queue) but I don't know if it;s worth the extra complexity - I assume that would be a rare case.

Reply:
nah, not worth the complication.